java/src/Ice/src/main/java/com/zeroc/Ice/ObjectAdapter.java

```diff
@@ -485,5 +943,746 @@ public interface ObjectAdapter {
    * @see #refreshPublishedEndpoints
    * @see Endpoint
    */
-  void setPublishedEndpoints(Endpoint[] newEndpoints);
+  public void setPublishedEndpoints(Endpoint[] newEndpoints) {
+    com.zeroc.IceInternal.LocatorInfo locatorInfo = null;
+    EndpointI[] oldPublishedEndpoints;
+
+    synchronized (this) {
+      checkForDeactivation();
+      if (_routerInfo != null) {
+        throw new IllegalArgumentException(
+            "can't set published endpoints on object adapter associated with a router");
+      }
+
+      oldPublishedEndpoints = _publishedEndpoints;
+      _publishedEndpoints =
+          java.util.Arrays.copyOf(newEndpoints, newEndpoints.length, EndpointI[].class);
+      locatorInfo = _locatorInfo;
+    }
+
+    try {
+      Identity dummy = new Identity();
+      dummy.name = "dummy";
+      updateLocatorRegistry(locatorInfo, createDirectProxy(dummy));
+    } catch (LocalException ex) {
+      synchronized (this) {
+        //
+        // Restore the old published endpoints.
+        //
+        _publishedEndpoints = oldPublishedEndpoints;
+        throw ex;
+      }
+    }
+  }
+
+  public boolean isLocal(ObjectPrx proxy) {
```

Comment:
A downside of this PR is various "internal" methods are now public on ObjectAdapter. Previously, they were also public but on ObjectAdapterI.

All or most of these internal methods need to be public because they are  called by code in the IceInternal package.